lib/scripts.js

```diff
@@ -124,7 +124,12 @@ var scripts = {
 
   moveToFinished: function(job, val, propVal, shouldRemove, target, ignoreLock){
     var args = scripts.moveToFinishedArgs(job, val, propVal, shouldRemove, target, ignoreLock);
-    return job.queue.client.moveToFinished(args);
+    return job.queue.client.moveToFinished(args).then(function(result){
+      switch(result){
+        case -1: throw Error('Missing key for job ' + job.id);
+        case -2: throw Error('Missing lock for job ' + job.id);
```

Comment:
yes, the error is not being propagated, it is a bit tricky though since it is the script is executed in a multi command. Basically what is needed is to first mark that we run moveToFinished and then pick the last result from multi. Will work on it now.
https://github.com/OptimalBits/bull/blob/movetofinished-errors/lib/job.js#L186-L189